### ISSUE TITLE
num_val_sample_per_class 

### ISSUE THREAD

**Issue by sab148** [clovaai/wsolevaluation#12]
Hi, I wanted to ask about the argument `num_val_sample_per_class ` in 
`python main.py --dataset_name OpenImages \
               --architecture vgg16 \
               --wsol_method cam \
               --experiment_name OpenImages_vgg16_CAM \
               --pretrained TRUE \
               --num_val_sample_per_class 5 \
               --large_feature_map FALSE \
               --batch_size 32 \
               --epochs 10 \
               --lr 0.00227913316 \
               --lr_decay_frequency 3 \
               --weight_decay 5.00E-04 \
               --override_cache FALSE \
               --workers 4 \
               --box_v2_metric True \
               --iou_threshold_list 30 50 70 \
               --eval_checkpoint_type last `

You set it to 5 for OpenImage dataset. It shouldn't be 25 instead because we have 25 sample per class for the validation set ? 

So when we will use your code with CUB and ImageNet datasets to which number should we set the argument `num_val_sample_per_class ` ?

Thank you in advance for you replay :)

**Comment by coallaoh** [comment]
Thanks for the question!

<img width="812" alt="Screen Shot 2020-05-02 at 10 11 00 AM" src="https://user-images.githubusercontent.com/7447092/80851516-3ce96f80-8c5d-11ea-83ea-e584e5e65ff4.png">

The `num_val_sample_per_class` should depend on which numbers you wish to compare against. For example, Table 2 of the main paper above is based on (10, 5, 5) validation samples for (ImageNet, CUB, OpenImages), where the caveat for CUB is that there are certain classes with <5 samples, so you should actually set `num_val_sample_per_class=-1` to indicate that you want all available validation samples ;) For other datasets, if you wish to compare against numbers in Table 2, follow the above criteria.

**Comment by sab148** [comment]
> Thanks for the question!
> 
> <img alt="Screen Shot 2020-05-02 at 10 11 00 AM" width="812" src="https://user-images.githubusercontent.com/7447092/80851516-3ce96f80-8c5d-11ea-83ea-e584e5e65ff4.png">
> 
> The `num_val_sample_per_class` should depend on which numbers you wish to compare against. For example, Table 2 of the main paper above is based on (10, 5, 5) validation samples for (ImageNet, CUB, OpenImages), where the caveat for CUB is that there are certain classes with <5 samples, so you should actually set `num_val_sample_per_class=-1` to indicate that you want all available validation samples ;) For other datasets, if you wish to compare against numbers in Table 2, follow the above criteria.

Thank you so much for you explanation and Bravo for your amazing work :)
